pkg/agent/client/client.go

```diff
@@ -144,22 +118,36 @@ func (c *Client) Revdial(ctx context.Context, path string) (*websocket.Conn, *ht
 }
 
 func (c *Client) get(ctx context.Context, out interface{}, s ...string) error {
-	req, err := http.NewRequest("GET", getURL(c.url, s...), nil)
+	bytes, err := c.getB(ctx, out, s...)
 	if err != nil {
 		return err
 	}
+	if len(bytes) == 0 {
+		return nil
+	}
+	return json.Unmarshal(bytes, &out)
+}
+
+func (c *Client) getB(ctx context.Context, out interface{}, s ...string) ([]byte, error) {
+	req, err := http.NewRequest("GET", getURL(c.url, s...), nil)
+	if err != nil {
+		return nil, err
+	}
 
 	req.SetBasicAuth(c.accessKey, "")
 
 	resp, err := c.httpClient.Do(req)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return nil, errors.WithMessagef(ErrNonSuccessResponse, "code %d", resp.StatusCode)
```

Comment:
Let's include the body of the response into the error and then centralize the processing of this logic into `pkg/http`.